mlos_bench/mlos_bench/config/schemas/services/remote/azure/azure-auth-service-subschema.json

```diff
@@ -23,10 +23,35 @@
                         "tokenRequestInterval": {
                             "description": "Time before token expiration to request a new token",
                             "type": "number"
+                        },
+                        "spClientId": {
+                            "description": "Client ID of the Service Principal to authenticate as",
+                            "type": "string"
+                        },
+                        "keyVaultName": {
+                            "description": "Name of key vault to retrieve the certificate from, for authorizing the Service Principal",
+                            "type": "string"
+                        },
+                        "certName": {
```

Comment:
Just a note: we might want to support `secretName` for pulling secrets as passwords instead, but I think for now only supporting certs is fine.